### ISSUE TITLE
Starting a transaction while another one is in progress locks up the client

### ISSUE THREAD

**Issue by davidisaaclee** [DallasHoff/sqlocal#45]
The following code never terminates (and all future queries hang):
```ts
await Promise.all([
  transaction(async (tx) => {
    console.log("txn 1:", await tx.sql`select * from groceries`);
    await sleep(100); // just to be sure the txns overlap
  }),
  transaction(async (tx) => {
    console.log("txn 2:", await tx.sql`select * from groceries`);
  }),
]);
```
[full gist](https://gist.github.com/davidisaaclee/44d70598ee4acbd98c4708961e50aade)

My reading is that:
1. The first transaction that gets to the worker works as expected: it's a new transaction when no transaction is active, so `partOfTransaction === true`, and it does a single lock / unlock of `transactionMutex` (starting [here](https://github.com/DallasHoff/sqlocal/blob/305374757bf300edb8ecaaa3203235cee19eb08d/src/processor.ts#L180))
2. The second transaction is noted as being `!partOfTransaction`, since there's an separate active transaction (`Processor#transactionKey != null && Processor#transactionKey != message.transaction`). It hits [the outer `transactionMutex.lock()`](https://github.com/DallasHoff/sqlocal/blob/305374757bf300edb8ecaaa3203235cee19eb08d/src/processor.ts#L152-L154) and waits for the first txn to finish.
3. Once the first txn finishes + releases the lock, second transaction continues, but **still has `transactionMutex` locked** – then it hits [the inner `transactionMutex.lock()`](https://github.com/DallasHoff/sqlocal/blob/305374757bf300edb8ecaaa3203235cee19eb08d/src/processor.ts#L180), and deadlocks.

My use case is that I'm inserting rows in transactions as objects come in over the network. It's understandable if your suggestion is just for me to use a queue, but I'd think that this should just work (or at least error).

**Comment by DallasHoff** [comment]
Thanks for the report! Yeah, that's definitely a bug. I'll look into it more later today. In the meantime, if you wanted to open a PR that adds your example as a failing unit test, that would be good.

**Comment by davidisaaclee** [comment]
Thanks for looking – opened #46 with the failing test.

**Comment by DallasHoff** [comment]
I appreciate it. I just published version 0.11.3 with the fix for this.
